### ISSUE TITLE
Unwanted behavior from executeOnText's warnIgnored

### ISSUE THREAD

**Issue by joshkel** [eslint/eslint#6611]
#6305 appears to have a bug or unintended behavior.

According to [docs/developer-guide/nodejs.md](https://github.com/eslint/eslint/blob/master/docs/developer-guide/nodejs-api.md#executeontext):

> If a filename in the optional second parameter matches a file that is configured to be ignored, then this function returns no errors or warnings. To return a warning instead, call the method with true as the optional third parameter.

Based on that, and based on the parameter name `warnIgnored`, I would expect that, if a file is configured to be ignored, then it either is ignored completely (`warnIgnored` is falsy, the default), or it generates a single warning telling the user that it's ignored (`warnIgnored` is truthy).

Instead, as far as I can tell, if `warnIgnored` is falsy, the file is not ignored; it's processed and generates warnings and errors just as if it weren't ignored at all. (See [here](https://github.com/eslint/eslint/blob/6d40d859cd09446de3721695f10ce75740507a6d/lib/cli-engine.js#L745).)

This resulted in MoOx/eslint-loader#97 with ESLint 3.0.

**Comment by platinumazure** [comment]
Are you suggesting that those CLIEngine lines should make look more like this?

```js
        if (filename && ignoredPaths.contains(filename)) {
            if (warnIgnored) {
                results.push(createIgnoreResult(filename, options.cwd));
            }
        } else {
            results.push(processText(text, configHelper, filename, options.fix, options.allowInlineConfig));
        }
```

**Comment by platinumazure** [comment]
@joshkel Awesome. I think you're probably correct that this is a bug. We'll see if the rest of the team agrees.

**Comment by nzakas** [comment]
Yeah, I think the logic is messed up. 

@joshkel in the future, please be sure to use our issue template to report bugs. It really helps us a lot.

**Comment by joshkel** [comment]
@nzakas - Will do. Sorry for the inconvenience.
